x-pack/plugins/security_solution/public/hosts/components/host_risk_score_table/severity_filter_group.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback, useMemo, useState } from 'react';
+
+import {
+  EuiFilterButton,
+  EuiFilterGroup,
+  EuiFilterSelectItem,
+  EuiPopover,
+  FilterChecked,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { useDispatch } from 'react-redux';
+import { HostRiskSeverity } from '../../../../common/search_strategy';
+import * as i18n from './translations';
+import { hostsActions, hostsModel, hostsSelectors } from '../../store';
+import { SeverityCount } from '../../containers/kpi_hosts/risky_hosts';
+import { HostRiskScore } from '../common/host_risk_score';
+import { useDeepEqualSelector } from '../../../common/hooks/use_selector';
+import { State } from '../../../common/store';
+
+interface SeverityItems {
+  risk: HostRiskSeverity;
+  count: number;
+  checked?: FilterChecked;
+}
+export const SeverityFilterGroup: React.FC<{
+  severityCount: SeverityCount;
+  type: hostsModel.HostsType;
+}> = ({ severityCount, type }) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const dispatch = useDispatch();
+
+  const onButtonClick = useCallback(() => {
+    setIsPopoverOpen(!isPopoverOpen);
+  }, [isPopoverOpen]);
+
+  const closePopover = useCallback(() => {
+    setIsPopoverOpen(false);
+  }, []);
+
+  const filterGroupPopoverId = useGeneratedHtmlId({
+    prefix: 'filterGroupPopover',
+  });
+  const getHostRiskScoreFilterQuerySelector = useMemo(
+    () => hostsSelectors.hostRiskScoreSeverityFilterSelector(),
+    []
+  );
+  const severitySelectionRedux = useDeepEqualSelector((state: State) =>
+    getHostRiskScoreFilterQuerySelector(state, type)
+  );
+
+  const items: SeverityItems[] = useMemo(() => {
+    const checked: FilterChecked = 'on';
```

Comment:
Do we really need this `const`?

Reply:
for the type, yes

Reply:
the type is checked in `const items: SeverityItems[]`, we can do: 
```checked: severitySelectionRedux.includes(k) ? 'on' : undefined```

Reply:
hm it was throwing error the other day when i did that but not now, i mustve changed something. i have a follow up pr coming after this, i will fix it there